examples/rfmip-clear-sky/rrtmgp_rfmip_sw.F90

```diff
@@ -265,25 +268,27 @@ program rrtmgp_rfmip_sw
 #endif
     ! Boundary conditions
     !   (This is partly to show how to keep work on GPUs using OpenACC in a host application)
-    !
     ! What's the total solar irradiance assumed by RRTMGP?
-    !  The first two loops could be more expressed more ompactly as def_tsi(1:block_size) = sum(toa_flux, dim=2)
     !
-    !!!$acc parallel loop
-    do icol = 1, block_size
-      def_tsi(icol) = toa_flux(icol, 1)
-    end do
-    !!!$acc parallel loop collapse(2)
+#ifdef _OPENACC
+    call zero_array(block_size, def_tsi)
+    !$acc parallel loop collapse(2) copy(def_tsi) copyin(toa_flux)
     do igpt = 1, ngpt
       do icol = 1, block_size
-        !!$acc atomic update
+        !$acc atomic update
         def_tsi(icol) = def_tsi(icol) + toa_flux(icol, igpt)
       end do
     end do
+#else
+    !
+    ! More compactly...
+    !
+    def_tsi(1:block_size) = sum(toa_flux, dim=2)
+#endif
```

Comment:
It might make sense to write your own "sum" at some point that does the `#ifdef` branching itself rather than doing it at the higher level like you do here. You can then create an OpenACC sum and a CPU sum under the hood.

Reply:
Yes, that's a good idea. I could use this in other contexts.